Content.Client/AME/AMEFuelContainerVisualizerSystem.cs

```diff
@@ -0,0 +1,25 @@
+using Content.Client.AME.Components;
+using Content.Shared.AME;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.AME;
+public sealed class AMEFuelContainerVisualizerSystem : VisualizerSystem<AMEFuelContainerVisualsComponent>
+{
+    protected override void OnAppearanceChange(EntityUid uid, AMEFuelContainerVisualsComponent component,
+        ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+
+        if (args.Component.TryGetData(AMEFuelContainerVisuals.IsOpen, out bool isOpen))
+        {
+            args.Sprite.LayerSetVisible(AMEFuelContainerVisualLayers.Seal, !isOpen);
+        }
+    }
+
+    public enum AMEFuelContainerVisualLayers
+    {
+        Seal
+    }
```

Comment:
this is super easy to just convert into a generic visualizer and remove all of the visualsystem/component boilerplate (all that needs to stay is the appearance data setting)

in the yaml you can do something like this:

```yaml

... # in sprite comp
    layers:
    - state: jar
    - state: seal
      map: ["seal"] # replace with just string layer key
      visible: true
... # further down

- type: Appearance
- type: GenericVisualizer
  visuals:
    enum.AMEFuelContainerVisuals.IsOpen:
      seal:
        True: { visible: false }
        False: { visible: true }
```

Reply:
note 4 myself: https://docs.spacestation14.io/en/engine/porting-visualizers